src/ui/CustomSelector/CustomSelector.tsx

```diff
@@ -3,27 +3,31 @@ import React, { useState } from 'react';
 import ArrowDownIcon from '@assets/icons/Caret_Down_MD.svg?react';
 import ArrowUpIcon from '@assets/icons/Caret_Up_MD.svg?react';
 
-import styles from './CustomSelector.module.css';
+import type { Selector } from '@/components/SearchBar/SearchBar';
+import { useFilterContext } from '@/hooks/useFilterContext';
 
-interface Selector {
-    title: string;
-    selector: string;
-}
+import styles from './CustomSelector.module.css';
 
 interface CustomSelectorProps {
-    selectors: Selector[];
+    selectors: Readonly<Selector[]>;
     title: string;
 }
 
 function CustomSelector({ selectors, title }: CustomSelectorProps) {
-    const [currentSelector, setCurrentSelector] = useState(selectors[0]);
+    const {
+        filtersState: { sortBy },
+        handleSelectFilter,
+    } = useFilterContext();
+
+    const [currentSelector, setCurrentSelector] = useState(selectors.find((sel) => sel.selector === sortBy));
```

Comment:
it is nice place for arrow function inside hook init
```suggestion
    const [currentSelector, setCurrentSelector] = useState(() => selectors.find((sel) => sel.selector === sortBy));
```